tests/tests_e3/python/main_test.py

```diff
@@ -149,3 +150,76 @@ def test_star_requirements():
         pypi.add_wheel(wheel3.path)
         pypi.add_requirement("src2==1.0.0")
         assert len(pypi.closure()) == 2
+
+
+def test_yanked(pypi_server):
```

Comment:
Maybe tests could be simpler using [pytest.mark.parametrize](https://docs.pytest.org/en/7.1.x/reference/reference.html#id23)

Reply:
I don't understand what's in it for us (or even how to use it in this case).

Can you explain what you have in mind?

Reply:
I think the `requests-mock` package should be added to `setup.py`'s test dependencies too.
```python
@pytest.mark.parametrize(
    "arguments,expected",
    [
        ((None, None), "setuptools_scm-7.1.0-py3-none-any.whl"),
        ((["setuptools-scm"], None), "setuptools_scm-8.0.0-py3-none-any.whl"),
        ((["setuptools_scm"], None), "setuptools_scm-8.0.0-py3-none-any.whl"),
        ((None, "setuptools_scm==8"), None)
    ]
)
def test_yanked(pypi_server, arguments, expected):
    allowed_yanked, invalid_wheel = arguments
    expected_wheel = expected

    with pypi_server:
        mkdir("cache")

        with PyPIClosure(
            python3_version=11,
            platforms=[
                "x86_64-linux",
            ],
            cache_dir="cache",
            cache_file="pypi.cache",
            allowed_yanked=allowed_yanked,
        ) as pypi:
            if invalid_wheel:
                with pytest.raises(
                    PyPIError,
                    match=(
                        "Cannot find latest version for 'setuptools-scm': "
                        "No more suitable version"
                    ),
                ):
                    pypi.add_requirement(invalid_wheel)
            else:
                pypi.add_requirement("setuptools_scm >= 6.2, <= 8")
                all_filenames = [basename(f) for f in pypi.file_closure()]
                assert expected in all_filenames
```